source/jormungandr/jormungandr/street_network/taxi.py

```diff
@@ -165,13 +165,17 @@ def _create_additional_section(self, pt_object, additional_time, begin_date_time
 
         return additional_section
 
-    def _update_journey(self, journey, additional_section):
+    def _update_journey(self, journey, additional_section, is_add_after):
         journey.duration += additional_section.duration
         journey.durations.total += additional_section.duration
         journey.arrival_date_time += additional_section.duration
 
-        # The sections will be sorted afterwards based on begin and end_date_time
-        journey.sections.extend([additional_section])
+        if is_add_after:
+            journey.sections.extend([additional_section])
+        else:
+            journey.sections.extend([journey.sections[0]])
```

Comment:
Shouldn't you just do (maybe needs copy of journey.sections before):
```python
journey.sections[:] = [additional_section] + journey.sections
```

Reply:
Right! Let's try that!

Reply:
Seems like you can't copy a full `RepeatedCompositeFieldContainer` into an other one.